docarray/array/doc_dict/doc_dict.py

```diff
@@ -0,0 +1,131 @@
+from typing import TYPE_CHECKING, Any, Dict, Iterable, Mapping, Type, TypeVar, Union
+
+import orjson
+from typing_inspect import is_union_type
+
+from docarray.array.any_collections import AnyCollection
+from docarray.array.doc_list.doc_list import DocList
+from docarray.base_doc import AnyDoc
+from docarray.base_doc.doc import BaseDoc
+from docarray.base_doc.io.json import orjson_dumps
+
+if TYPE_CHECKING:
+    from docarray.proto import DocDictProto
+    from docarray.typing import NdArray, TorchTensor
+
+T_doc = TypeVar('T_doc', bound=BaseDoc)
+T = TypeVar('T', bound='DocDict')
+
+
+class DocDict(AnyCollection[T_doc], Dict[str, T_doc]):
+    doc_type: Type[BaseDoc] = AnyDoc
+
+    def __init__(self, **mapping):
+        super().__init__(
+            {key: self._validate_one_doc(val) for key, val in mapping.items()}
+        )
```

Comment:
Didn't we want to have a default where if there is no key given, we use the id as key?

Reply:
it is define in the `from_doc_list` method.

The init just take a mapping, like a normal dict

Reply:
I think normal dict takes more, no? Like list of tuples for example? Or am I wrong?

Reply:
let me double check, but I thin the `dict()` function can just take a mapping. I will just follow whatever dict init do, Makes sense to you ?